contracts/dao/TribeMinter.sol

```diff
@@ -0,0 +1,211 @@
+    }
+
+    /// @notice add an address to the lockedTribe excluded list
+    function addLockedTribeAddress(address lockedTribeAddress) external override onlyGovernorOrAdmin {
+        _addLockedTribeAddress(lockedTribeAddress);
+    }
+
+    /// @notice remove an address from the lockedTribe excluded list
+    function removeLockedTribeAddress(address lockedTribeAddress) external override onlyGovernorOrAdmin {
+        _lockedTribeAddresses.remove(lockedTribeAddress);
+        emit RemoveLockedTribeAddress(lockedTribeAddress);
+    }
+
+    /// @notice changes the TRIBE minter address
+    /// @param newMinter the new minter address
+    function setMinter(address newMinter) external override onlyOwner {
+        require(newMinter != address(0), "TribeReserveStabilizer: zero address");
+        ITribe _tribe = ITribe(address(tribe()));
+        _tribe.setMinter(newMinter);
+    }
+
+    /// @notice sets the max annual inflation relative to current supply
+    /// @param newAnnualMaxInflationBasisPoints the new max inflation % denominated in basis points (1/10000)
+    function setAnnualMaxInflationBasisPoints(uint256 newAnnualMaxInflationBasisPoints) external override onlyGovernorOrAdmin {
+        _setAnnualMaxInflationBasisPoints(newAnnualMaxInflationBasisPoints);
+    }
+
+    /// @notice return the ideal buffer cap based on TRIBE circulating supply
+    function idealBufferCap() public view override returns (uint256) {
+        return tribeCirculatingSupply() * annualMaxInflationBasisPoints / Constants.BASIS_POINTS_GRANULARITY;
+    }
+
+    /// @notice return the TRIBE supply, subtracting locked TRIBE
+    function tribeCirculatingSupply() public view override returns (uint256) {
+        IERC20 _tribe = tribe();
+
+        // Remove all locked TRIBE from total supply calculation
+        uint256 lockedTribe;
+        for (uint256 i = 0; i < _lockedTribeAddresses.length(); i++) {
+            lockedTribe += _tribe.balanceOf(_lockedTribeAddresses.at(i));
+        }
+
+        return _tribe.totalSupply() - lockedTribe;
+    }
+
+    /// @notice alias for tribeCirculatingSupply
+    /// @dev for compatibility with ERC-20 standard for off-chain 3rd party sites
+    function totalSupply() public view override returns (uint256) {
+        return tribeCirculatingSupply();
+    }
+
+    /// @notice return whether a poke is needed or not i.e. is buffer cap != ideal cap
+    function isPokeNeeded() external view override returns (bool) {
+        return idealBufferCap() != bufferCap;
+    }
+
+    /// @notice return the set of locked TRIBE holding addresses to be excluded from circulating supply
+    function lockedTribeAddresses() external view override returns(address[] memory) {
+        return _lockedTribeAddresses.values();
+    }
+
+    function _addLockedTribeAddress(address lockedTribeAddress) internal {
+        _lockedTribeAddresses.add(lockedTribeAddress);
+        emit AddLockedTribeAddress(lockedTribeAddress);
```

Comment:
only emit an event if .add returns true; else the address was already in the enumerableSet

Reply:
no longer using locked addresses